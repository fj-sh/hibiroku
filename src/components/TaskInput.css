.TaskInput_container {
  width: 100%;
  border-bottom: #dcdcdc;
  margin: 0 0 2rem 0;
  box-shadow: rgba(0, 0, 0, 0.1) 0px 4px 12px;
}
.TaskInput_input {
  padding: 10px;
  margin: 10px;
  height: 20px;
  width: 350px;
  border:1px solid #eaeaea;
  border-radius: 10px;
  outline:none;
}

.TaskInput_input:hover {
  border-color: #a0a0a0 #b9b9b9 #b9b9b9 #b9b9b9;
}

.TaskInput_input:focus {
  border-color:#4d90fe;
}

.TaskInput_button_add {
  border-radius: 5px;
  background: #1490f4;
  border: 1px solid #f2f2f2;
  color: #f5f5f5;
  cursor: default;
  font-size: 14px;
  font-weight: bold;
  max-width: 60px;
  padding: 0 16px;
  height:36px;
}

.TaskInput_button_add:hover {
  box-shadow: 0 1px 1px rgba(0,0,0,0.1);
  background: #f8f8f8;
  border: 1px solid #c6c6c6;
  color: #222;
}
