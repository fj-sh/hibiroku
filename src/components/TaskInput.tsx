import React, { useRef } from 'react'
import { randomId } from '../lib/uuid'
import { Task } from '../lib/types'
import './TaskInput.css'

type Props = {
  onAdd: (taskName: Task) => void
}

const TaskInput = ({ onAdd }: Props) => {
  const inputEl = useRef<HTMLInputElement>(null)

  const handleAddTask = () => {
    if (inputEl.current.value === '') return
    const task: Task = {
      id: randomId(),
      name: inputEl.current.value,
      status: 'Stopped',
      totalSeconds: 0,
    }
    onAdd(task)
    inputEl.current.value = ''
  }

  const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
    if (e.nativeEvent.isComposing || e.key !== 'Enter') return
    handleAddTask()
  }

  return (
    <div className="TaskInput_container">
      <label>
        <input
          data-testid="TaskInput-input"
          type="text"
          name="taskName"
          placeholder="Enter your task"
          className="TaskInput_input"
          onKeyDown={handleKeyDown}
          ref={inputEl}
        />
        <button onClick={handleAddTask} className="TaskInput_button_add">
          Add
        </button>
      </label>
    </div>
  )
}

export default TaskInput
