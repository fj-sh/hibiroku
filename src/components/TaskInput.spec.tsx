import React from 'react'
import renderer from 'react-test-renderer'
import TaskInput from './TaskInput'

describe('TaskInput', () => {
  it('Create snapshot', () => {
    const onAdd = () => {}

    const component = renderer.create(<TaskInput onAdd={onAdd} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
