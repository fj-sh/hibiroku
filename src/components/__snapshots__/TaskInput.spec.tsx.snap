// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskInput Create snapshot 1`] = `
<div>
  <label>
    Register a Task:
    <input
      name="taskName"
      onKeyDown={[Function]}
      placeholder="Input your task"
      type="text"
    />
    <button
      onClick={[Function]}
    >
      Add
    </button>
  </label>
</div>
`;
